crates/csharp/src/function.rs

```diff
@@ -1027,6 +1114,24 @@ impl Bindgen for FunctionBindgen<'_, '_> {
                     }
                 }
 
+                if is_async {
+                    self.interface_gen.csharp_gen.needs_async_support = true;
+                    let name = self.func_name.to_upper_camel_case();
+                    let ret_param = match func.result {
+                            None => "",
+                            Some(_ty) => "ret.Result"
+                            };
+
+                    uwriteln!(self.src, r#"if (ret.IsCompletedSuccessfully) 
+                    {{
+                        {name}TaskReturn({ret_param});
+                        return (uint)CallbackCode.Exit;
+                    }}
+                    
+                    return (uint)CallbackCode.Yield;
+                    "#);
+                }
+
```

Comment:
Might want to add a "TODO" comment here to remember to defer dropping borrowed resources until a result is returned (i.e. just before calling `{name}TaskReturn`).

Reply:
Thanks, done